src/Polly.Core/Simmy/Latency/LatencyChaosStrategy.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Latency;
+
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+internal sealed class LatencyChaosStrategy : MonkeyStrategy
+{
+    private readonly TimeProvider _timeProvider;
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public LatencyChaosStrategy(
+        LatencyStrategyOptions options,
+        TimeProvider timeProvider,
+        ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        Guard.NotNull(telemetry);
+        Guard.NotNull(timeProvider);
+
+        if (options.Latency is null && options.LatencyGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Latency), "Either Latency or LatencyGenerator is required.");
+        }
+
+        Latency = options.Latency;
+        LatencyGenerator = options.Latency.HasValue ? (_) => new(options.Latency.Value) : options.LatencyGenerator;
+        OnDelayed = options.OnDelayed;
+
+        _telemetry = telemetry;
+        _timeProvider = timeProvider;
+    }
+
+    public Func<OnDelayedArguments, ValueTask>? OnDelayed { get; }
+
+    public Func<ResilienceContext, ValueTask<TimeSpan>> LatencyGenerator { get; }
+
+    public TimeSpan? Latency { get; }
+
+    protected internal override async ValueTask<Outcome<TResult>> ExecuteCore<TResult, TState>(
+        Func<ResilienceContext, TState, ValueTask<Outcome<TResult>>> callback, ResilienceContext context, TState state)
+    {
+        try
+        {
+            if (await ShouldInject(context).ConfigureAwait(context.ContinueOnCapturedContext))
+            {
+                var latency = await LatencyGenerator(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                await _timeProvider.DelayAsync(latency, context).ConfigureAwait(context.ContinueOnCapturedContext);
```

Comment:
Can we use the same approach as for retry when `OnRetry` is called before next delay?

https://github.com/App-vNext/Polly/blob/b1ec86347a5a2fe163d484a6bc19e33bbf804bf8/src/Polly.Core/Retry/RetryResilienceStrategy.cs#L77

In this case we would call and report the event and only then call delay.

Reply:
What happens then if the delay is canceled or even if it throws another kind of exception? we would report an event that never happened 🤔

Reply:
> we would report an event that never happened 

We are just reporting that delay is about to happen. (same with retry).
If someone cancels it's ok, but we at least the resilience event was recorded.